phoenix-core-client/src/main/java/org/apache/phoenix/util/UpgradeUtil.java

```diff
@@ -1579,6 +1588,12 @@ public static void moveHBaseLevelTTLToSYSCAT(PhoenixConnection oldMetaConnection
                                         "in progress => numOfTableHasTTLMoved: %d",
                                 numOfTableThatHasTTLMoved));
 
+                    } catch (SQLException e) {
+                        LOGGER.error(String.format(
+                                "Failed moving ttl value to TTL" +
+                                        " column for %s with Exception :",
+                                currentTableName), e);
+
```

Comment:
Also, to log the error for any exception (e.g. NPE that we came across), how about catching generic exception and logging the current table name?

```
                    } catch (Exception e) {
                        LOGGER.error("Failed moving ttl value to TTL"
                                + " column for {} with Exception :", currentTableName, e);
                        throw new RuntimeException(e);
                    }
```

Reply:
Eventually top level caller `CQSI#upgradeSystemTables()` anyways take care of wrapping any non-SQL exceptions into SQLException.